### PR TITLE
Improve internal types

### PR DESCRIPTION
**Pull request by SferaDev**
- Filter values should not show internal methods. Closes: https://github.com/xataio/client-ts/issues/6
- Make codegen fields ``nullable``
- Improve types for basic ``deno`` support

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -135,3 +135,20 @@ await user.delete();
 // Using an object's id
 await xata.db.users.delete('rec_1234abcdef');
 ```
+
+## Deno support
+
+Right now we are still not publishing the client on deno.land or have support for deno in the codegen.
+
+However you can already use it with your preferred node CDN with the following import in the auto-generated ``:
```
Comment (gimenete) [inline README.md:143]: Is something missing between these backticks?

**Review by gimenete — APPROVED** [review]
LGTM. Just a minor comment.

File: README.md (hunk 1 above)
Comment (SferaDev) [inline README.md:143]:
```suggestion
However you can already use it with your preferred node CDN with the following import in the auto-generated ``xata.ts`` file:
```

Comment (SferaDev) [reply README.md:143]: Actually yes 🤯

File: README.md [hunk 2]
```diff
@@ -135,3 +135,20 @@ await user.delete();
 // Using an object's id
 await xata.db.users.delete('rec_1234abcdef');
 ```
+
+## Deno support
```
Comment (gimenete) [inline README.md:139]:
```suggestion
## Deno support
```

There's a weird whitespace that prevents this line to be rendered as a heading. See https://github.com/xataio/client-ts/blob/f1e660da2d79ab2314266f0fb2e18055e944bbe2/README.md

File: README.md [hunk 3]
```diff
@@ -135,3 +135,20 @@ await user.delete();
...
+
+Right now we are still not publishing the client on deno.land or have support for deno in the codegen.
+
+However you can already use it with your preferred node CDN with the following import in the auto-generated ``xata.ts`` file:
```
Comment (gimenete) [inline README.md:143]:
```suggestion
However you can already use it with your preferred node CDN with the following import in the auto-generated `xata.ts` file:
```

Maybe you put two double ticks on purpose? But I don't think it's necessary.

**Pull request merged**